Documentation/cmdref/cilium-agent.md

```diff
@@ -85,6 +85,7 @@ cilium-agent [flags]
       --enable-host-port                                     Enable k8s hostPort mapping feature (requires enabling enable-node-port) (default true)
       --enable-host-reachable-services                       Enable reachability of services for host applications
       --enable-hubble                                        Enable hubble server
+      --enable-hubble-recorder-api                           Enable the Hubble recorder API
```

Comment:
One small nit, but can be done as follow-up. I would love to avoid adding yet another flag here. Given we already need `--enable-recorder=true` and if users also have `--enable-hubble=true`, then this could also automatically imply that the hubble recorder api will be enabled. So I would just remove this additional flag, given it's more hassle for users to configure, wdyt?

Reply:
I think we can default the flag to true if the recorder is enabled, but I'd like to have a flag to turn off the API, since it's quite privileged.

Reply:
@gandro Ok, that works as well. In that sense the use case for `--enable-recorder=true` and `--enable-hubble-recorder-api=false` would be if someone would hook up their own agent to configure recorder objects for the datapath (e.g. using same API as the CLI) and then process the traffic from the perf RB instead of letting the agent do it. I presume the agent doesn't hook up anything to the perf RB w/o active users, right?

Reply:
Correct. Maybe to extend on that a bit:
I mainly added the `enable-hubble-recorder-api` because we might also want to add a `enable-hubble-observer-api=false` flag to disable the "main" Hubble API. Right now, users of the Recorder API also have to enable normal Hubble which potentially has some non-negligible overhead to the agent's memory and cpu consumption.

So just from the Hubble side alone it seems reasonable to me to maybe be able to turn off specific parts of Hubble if a user really cares about performance.